src/utils/RequestQueue.cc

```diff
@@ -17,6 +17,14 @@ RequestQueue::~RequestQueue() {
     delete [] requests;
 }
 
+bool RequestQueue::empty() {
+    bool answer;
+    request_queue_mutex.lock();
+    answer = (count == 0);
+    request_queue_mutex.unlock();
+    return answer;
```

Comment:
Would the cost of a `std::scoped_lock` allocation be worth it for a cleaner and safer code?
Or the allocation cost is not irrelevant here?

```suggestion
    std::scoped_lock lock(this->request_queue_mutex);
    return count == 0;
```

Reply:
RequestQueue is the synchronized queue used throughout the code in many different places where threads need to communicate. I'd like to keep it as efficient as possible. In this place you marked I believe std::scoped_lock is harmless but there are many other uses of different mutex in this code which would made the use of std::scope less natural and hence more costly. So I prefer to use the same semaphore construction all along the code instead of having different ways of doing the same thing.

Perhaps std::scoped_lock could be used in other classes down the stream in the code where we have only this simple setter/setter behavior as in the method you marked.